java/core/src/java/org/apache/orc/OrcConf.java

```diff
@@ -219,7 +219,10 @@ public enum OrcConf {
       "orc.proleptic.gregorian.default", false,
       "This value controls whether pre-ORC 27 files are using the hybrid or proleptic\n" +
       "calendar. Only Hive 3.1 and the C++ library wrote using the proleptic, so hybrid\n" +
-      "is the default.")
+      "is the default."),
+  ROW_BATCH_SIZE("orc.row.batch.size", "orc.row.batch.size", 1024,
```

Comment:
Do we need to split the two configurations? This way the reader and writer use different batch sizes.
I see the discussion result of [ORC-777](https://issues.apache.org/jira/browse/ORC-777) is not introducing configuration items to confuse users.

Reply:
cc @pgaref

Reply:
How about the orc.read.row.batch.size` configuration item in this PR?
Then I raise another PR about `NegativeArraySizeException`.
If the writer needs to support configuration in the future, I will raise another PR and configure it with `orc.write.row.batch.size`.